### PR TITLE
feat(ui5-menu): enable navigation over disabled items

### PR DESCRIPTION
**Pull request by unazko**
- There is a new behavior implemented, which enables
the end users to navigate via mouse and keyboard to a
disabled `ui5-menu-item` element. This behavior is aimed
to improve the accessibility of the `ui5-menu` component
by providing screen reader announcements for disabled
menu items as well.

- The haspopup attribute is now applied to the` ui5-menu item`'s
with a sub-menu.

- There is a new `tooltip` property for the `ui5-menu-item`.

Fixes: https://github.com/SAP/ui5-webcomponents/issues/7096
Fixes: https://github.com/SAP/ui5-webcomponents/issues/8214

### CODE REVIEW

File: packages/main/src/MenuItem.ts [hunk 1]
```diff
@@ -117,6 +117,16 @@ class MenuItem extends UI5Element {
 	@property()
 	accessibleName!: string;
 
+	/**
+	 * Defines the text of the tooltip that would be displayed for the menu item.
```
Comment (BorisDafov) [inline packages/main/src/MenuItem.ts:121]: I suggest the shorter:
Defines the text of the tooltip for the menu item.

**Review by BorisDafov — CHANGES_REQUESTED** [review]

**Review by NHristov-sap — APPROVED** [review]

File: packages/main/src/MenuItem.ts (hunk 1 above)
Comment (unazko) [reply packages/main/src/MenuItem.ts]: done

**Review by tsanislavgatev — DISMISSED** [review]

**Comment by tsanislavgatev** [comment]
I only have one question, should the focus be dimmed if the item is disabled?

**Review by dobrinyonkov — APPROVED** [review]

**Review by NHristov-sap — APPROVED** [review]

**Comment by unazko** [comment]
> I only have one question, should the focus be dimmed if the item is disabled?

I've redirected this question to the design colleagues as this case isn't specifically showcased in the visual design.
According to the focus visual concept the focus outline is always displayed without opacity.
I suggest to leave it at that for now as currently I've removed the opacity over the focus outline so the result is the same as in the openui5 control counterpart.

We can set opacity once again in a separate PR if needed when we receive feedback from the design colleagues.

File: packages/main/src/MenuListItem.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	tag: "ui5-menu-li",
+	template: StandardListItemTemplate,
+	styles: [StandardListItem.styles, menuListItemCss],
+})
+class MenuListItem extends StandardListItem {
+	/**
+	 * Defines the associated MenuItem instance
+	 * @private
+	 */
+	@property({ type: Object, multiple: false })
```
Comment (nnaydenow) [inline packages/main/src/MenuListItem.ts:29]: multiple: false is not needed

File: packages/main/src/MenuItem.ts [hunk 3]
```diff
@@ -117,6 +117,16 @@ class MenuItem extends UI5Element {
...
 	accessibleName!: string;
 
+	/**
+	 * Defines the text of the tooltip for the menu item.
+	 *
+	 * @default ""
+	 * @public
+	 * @since 1.23.0
+	 */
+	@property({ type: String, defaultValue: "" })
```
Comment (nnaydenow) [inline packages/main/src/MenuItem.ts:127]: defaultValue: "" is not needed

File: packages/main/src/MenuListItem.ts (hunk 2 above)
Comment (unazko) [reply packages/main/src/MenuListItem.ts:29]: done

File: packages/main/src/MenuItem.ts (hunk 3 above)
Comment (unazko) [reply packages/main/src/MenuItem.ts:127]: done

**Pull request merged**